sql/core/src/test/scala/org/apache/spark/sql/DatasetUnpivotSuite.scala

```diff
@@ -305,14 +305,17 @@ class DatasetUnpivotSuite extends QueryTest
           valueColumnName = "val"
         )
     }
-    checkErrorClass(
+    checkError(
       exception = e,
       errorClass = "UNPIVOT_VALUE_DATA_TYPE_MISMATCH",
-      msg = "Unpivot value columns must share a least common type, some types do not: \\[" +
-        "\"STRING\" \\(`str1#\\d+`\\), " +
-        "\"INT\" \\(`int1#\\d+`, `int2#\\d+`, `int3#\\d+`, ...\\), " +
-        "\"BIGINT\" \\(`long1#\\d+L`, `long2#\\d+L`\\)\\];(\n.*)*",
-      matchMsg = true)
+      errorSubClass = None,
+      sqlState = None,
```

Comment:
Does `checkError` provide default parameter value for `errorSubClass` and `sqlState`?

Reply:
It has but if I remove settings of the parameters, I am getting the errors:
```
overloaded method value checkError with alternatives:
  (exception: org.apache.spark.SparkThrowable,errorClass: String,parameters: Map[String,String])Unit <and>
  (exception: org.apache.spark.SparkThrowable,errorClass: String,sqlState: String,parameters: Map[String,String])Unit <and>
  (exception: org.apache.spark.SparkThrowable,errorClass: String,errorSubClass: String,sqlState: String,parameters: Map[String,String])Unit <and>
  (exception: org.apache.spark.SparkThrowable,errorClass: String,errorSubClass: Option[String],sqlState: Option[String],parameters: Map[String,String],matchPVals: Boolean)Unit
 cannot be applied to (exception: org.apache.spark.sql.AnalysisException, errorClass: String, parameters: scala.collection.immutable.Map[String,String], matchPVals: Boolean)
    checkError(
```

Reply:
Let me add the overloaded method:
```scala
 protected def checkError(
      exception: SparkThrowable,
      errorClass: String,
      parameters: Map[String, String],
      matchPVals: Boolean): Unit =
```

Reply:
but I think it is better to set default values in the most wide method. I will remove one overloaded method and add default values.